src/timeMachine/components/view_options/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {FC, useState} from 'react'
+import {useSelector} from 'react-redux'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+
+import {getVisTable} from 'src/timeMachine/selectors'
+
+import {useVisXDomainSettings} from 'src/shared/utils/useVisDomainSettings'
+
+interface Props {
+  setDomain: (xDomain: [number, number]) => void
+  xDomain: number[] | null
+}
+
+const isValidXDomain = xDomain => {
```

Comment:
this is an odd place for this function. does something like this in giraffe @TCL735? even if it doesn't, this seems like it belongs elsewhere.

does it need to happen? it feels like this function existing is related to `xDomain` being `number[] | null`...

Reply:
It doesn't need to happen per se. Giraffe will auto-calculate the domain if null is passed in or if any of the following is not given: xDomain, onSetXDomain, onResetXDomain.

But Scatterplot is trying to do a lot here. It wants to use the time range that appears in a dropdown (**not** in the View Options), so that it can handle something other than __time_ on the x-axis.

And the proposed changes here also allow Scatterplot to handle __time_ but ignore the dropdown and auto-calculate the _time domain. I guess we are passing in the __time_ domain specifically to be consistent with the other behavior described above.

If instead Scatterplot passes it sometimes but not other times, it might be confusing for future Devs. And the type checking here is just in support of being consistent.